addons/l10n_in/models/account_invoice.py

```diff
@@ -37,15 +37,18 @@ def _compute_l10n_in_gst_treatment(self):
         for record in self:
             record.l10n_in_gst_treatment = record.partner_id.l10n_in_gst_treatment
 
-    @api.depends('partner_id')
+    @api.depends('partner_id', 'company_id')
     def _compute_l10n_in_state_id(self):
         for move in self:
             if move.country_code == 'IN':
-                country_code = move.partner_id.country_id.code
-                if country_code == 'IN':
-                    move.l10n_in_state_id = move.partner_id.state_id
-                elif country_code:
-                    move.l10n_in_state_id = self.env.ref('l10n_in.state_in_oc', raise_if_not_found=False)
+                if move.journal_id.type == 'sale':
```

Comment:
Is not is simpler to do if move.country_code == 'IN' and move.journal_id.type == 'sale':

in the if above?

Reply:
@jco-odoo Done